js/adapt-contrib-bookmarking.js

```diff
@@ -1,235 +1,235 @@
+      });
+    },
+
+    checkRestoreLocation: function() {
+      this.restoredLocationID = Adapt.offlineStorage.get('location');
+
+      if (!this.restoredLocationID || this.restoredLocationID === 'undefined') return;
+
+      this.listenToOnce(Adapt, 'pageView:ready menuView:ready', this.restoreLocation);
+    },
+
+    restoreLocation: function() {
+      _.defer(function() {
+        this.stopListening(Adapt, 'pageView:ready menuView:ready', this.restoreLocation);
+
+        if ((this.restoredLocationID === Adapt.location._currentId) || !Adapt.findById(this.restoredLocationID)) {
+          return;
         }
+        var locationOnscreen = $('.' + this.restoredLocationID).onscreen();
+        var isLocationOnscreen = locationOnscreen && (locationOnscreen.percentInview > 0);
+        var isLocationFullyInview = locationOnscreen && (locationOnscreen.percentInview === 100);
+        if (isLocationOnscreen && isLocationFullyInview) {
+          return;
+        }
+
+        if (Adapt.course.get('_bookmarking')._showPrompt === false) {
+          this.navigateToPrevious();
+          return;
+        }
+        this.showPrompt();
+        
+      }.bind(this));
+    },
+
+    showPrompt: function() {
+      var courseBookmarkModel = Adapt.course.get('_bookmarking');
+      var buttons = courseBookmarkModel._buttons || { yes: 'Yes', no: 'No' };
+
+      this.listenToOnce(Adapt, {
+        'bookmarking:continue': this.navigateToPrevious,
+        'bookmarking:cancel': this.navigateCancel
+      });
+
+      var promptObject = {
+        title: courseBookmarkModel.title,
+        body: courseBookmarkModel.body,
+        _prompts:[
+          {
+            promptText: buttons.yes || 'Yes',
+            _callbackEvent: 'bookmarking:continue'
+          },
+          {
+            promptText: buttons.no || 'No',
+            _callbackEvent: 'bookmarking:cancel'
+          }
+        ],
+        _showIcon: true
+      };
+
+      var accessibility = Adapt.config.get('_accessibility');
+      if (!accessibility || !accessibility._isActive) {
+        Adapt.trigger('notify:prompt', promptObject);
+        return;
+      }
+
+      $('.js-loading').show();
+      $('#a11y-focuser').focus();
+      $('body').attr('aria-hidden', true);
+      _.delay(function() {
+        $('.js-loading').hide();
+        $('body').removeAttr('aria-hidden');
+        Adapt.trigger('notify:prompt', promptObject);
+      }, 3000);
+    },
+
+    navigateToPrevious: function() {
+      _.defer(function() {
+        var isSinglePage = Adapt.contentObjects.models.length == 1;
+        Backbone.history.navigate('#/id/' + this.restoredLocationID, {trigger: true, replace: isSinglePage});
+      }.bind(this));
+
+      this.stopListening(Adapt, 'bookmarking:cancel');
+    },
+
+    navigateCancel: function() {
+      this.stopListening(Adapt, 'bookmarking:continue');
+    },
+
+    resetLocationID: function () {
+      this.setLocationID('');
+    },
+
+    /*
```

Comment:
function comments should start with `/**`
```suggestion
    /**
```